spec/services/pipeline_report_service_spec.rb

```diff
@@ -0,0 +1,261 @@
+                             },])
+
+      @background = create(:background,
+                           pipeline_run_ids: [
+                             create(:pipeline_run,
+                                    sample: create(:sample, project: create(:project))).id,
+                             create(:pipeline_run,
+                                    sample: create(:sample, project: create(:project))).id,
+                           ],
+                           taxon_summaries_data: [{
+                             tax_id: 1313,
+                             count_type: "NR",
+                             tax_level: 1,
+                             mean: 81.3845,
+                             stdev: 404.076,
+                           }, {
+                             tax_id: 1313,
+                             count_type: "NT",
+                             tax_level: 1,
+                             mean: 81.6257,
+                             stdev: 442.207,
+                           }, {
+                             tax_id: 1301,
+                             count_type: "NR",
+                             tax_level: 2,
+                             mean: 201.318,
+                             stdev: 942.975,
+                           }, {
+                             tax_id: 1301,
+                             count_type: "NT",
+                             tax_level: 2,
+                             mean: 290.481,
+                             stdev: 1482.97,
+                           },])
+
+      @report = PipelineReportService.call(@pipeline_run.id, @background.id)
+    end
+
+    it "should get correct species values" do
+      species_result = {
+        "genus_tax_id" => 1301,
+        "name" => "Streptococcus pneumoniae",
+        "nt" => {
+          "count" => 0,
+          "rpm" => 0,
+          "z_score" => -100,
+          "e_value" => 0,
+        },
+        "nr" => {
+          "count" => 2,
+          "rpm" => 1782.5311942959001, # previously rounded to 1782.531
+          "z_score" => 4.209967170274651, # previously rounded to 4.2099668
+          "e_value" => 9.3,
+        },
+        "agg_score" => 12_583.634591815486 # previously rounded to 12_583.63
+      }
+
+      expect(JSON.parse(@report)["counts"]["1"]["1313"]).to include_json(species_result)
+    end
+
+    it "should get correct genus values" do
+      genus_result = {
+        "genus_tax_id" => 1301,
+        "nt" => {
+          "count" => 4.0,
+          "rpm" => 3565.0623885918003, # previously rounded to 3565.062
+          "z_score" => 2.208123824886411, # previously rounded to 2.2081236
+          "e_value" => 81.478,
+        },
+        "nr" => {
+          "count" => 2.0,
+          "rpm" => 1782.5311942959001, # previously rounded to 1782.531
+          "z_score" => 1.6768346926439197, # previously rounded to 1.6768345
+          "e_value" => 9.3,
+        },
+        "agg_score" => 73_603.777,
```

Comment:
The [test](https://github.com/chanzuckerberg/idseq-web/blob/master/test/controllers/samples_controller_test.rb#L176) expects a genus aggregate score of 73603.777, but the report returns 12583.634591815486 (same as the species aggregate score).

Reply:
Thanks for flagging this! @tfrcarvalho do you know why?

Reply:
This a bug in the new report service - we are still investigating.
This a draft PR to facilitate communication between Julie and I.

Reply:
Caught the bug, there was a typo in the test and the background model was missing some taxon summary data.